src/nvim/rbuffer.c

```diff
@@ -121,7 +121,7 @@ char *rbuffer_read_ptr(RBuffer *buf, size_t *read_count) FUNC_ATTR_NONNULL_ALL
 {
   if (!buf->size) {
     *read_count = 0;
-    return NULL;
+    return buf->read_ptr;
```

Comment:
Wouldn't they still need to check nonzero size? Otherwise will they not deref an initialized buffer instead in the situation it crashed?